libfwupdplugin/fu-plugin.c

```diff
@@ -1872,8 +1885,14 @@ fu_plugin_runner_udev_device_changed (FuPlugin *self, FuUdevDevice *device, GErr
 
 	/* optional */
 	g_module_symbol (priv->module, "fu_plugin_udev_device_changed", (gpointer *) &func);
-	if (func == NULL)
+	if (func == NULL) {
+		if (priv->device_gtype != G_TYPE_INVALID ||
+		    fu_device_get_specialized_gtype (FU_DEVICE (device)) != G_TYPE_INVALID) {
```

Comment:
do we need both? I was surprised to see the device_gtype one.

Reply:
I was modeling off of the usb ones, which yes do.